packages/circuits/test/proveReputation.test.ts

```diff
@@ -11,32 +11,206 @@ describe('Prove reputation from attester circuit', function () {
+        expect(data.attesterId.toString()).to.equal('1')
+        expect(data.proveMinRep.toString()).to.equal('1')
+        expect(data.proveMaxRep.toString()).to.equal('1')
+        expect(data.proveZeroRep.toString()).to.equal('0')
+        expect(data.minRep.toString()).to.equal('0')
+        expect(data.maxRep.toString()).to.equal('0')
+        expect(data.proveGraffiti.toString()).to.equal('0')
+        expect(data.graffitiPreImage.toString()).to.equal('0')
+    })
+
+    it('should prove a minRep and maxRep and zero rep', async () => {
+        const id = new ZkIdentity()
+        const circuitInputs = genReputationCircuitInput({
+            id,
+            epoch: 1,
+            nonce: 0,
+            attesterId: 1,
+            startBalance: { posRep: 10, negRep: 10, graffiti: 0, timestamp: 0 },
+            minRep: 0,
+            maxRep: 0,
+            proveMaxRep: 1,
+            proveMinRep: 1,
+            proveZeroRep: 1,
+            revealNonce: 1,
         })
-        const { isValid } = await genProofAndVerify(
+        const { isValid, publicSignals, proof } = await genProofAndVerify(
             Circuit.proveReputation,
             circuitInputs
         )
         expect(isValid).to.be.true
+        expect(publicSignals[2].toString()).to.equal(
+            ReputationProof.buildControlInput({
+                epoch: 1,
+                nonce: 0,
+                attesterId: 1,
+                revealNonce: 1,
+                proveGraffiti: 0,
+                minRep: 0,
+                maxRep: 0,
+                proveMaxRep: 1,
+                proveMinRep: 1,
+                proveZeroRep: 1,
+            })[0].toString()
+        )
+        const data = new ReputationProof(publicSignals, proof)
+        expect(data.epoch.toString()).to.equal('1')
+        expect(data.nonce.toString()).to.equal('0')
+        expect(data.revealNonce.toString()).to.equal('1')
+        expect(data.attesterId.toString()).to.equal('1')
+        expect(data.proveMinRep.toString()).to.equal('1')
+        expect(data.proveMaxRep.toString()).to.equal('1')
+        expect(data.proveZeroRep.toString()).to.equal('1')
+        expect(data.minRep.toString()).to.equal('0')
+        expect(data.maxRep.toString()).to.equal('0')
+        expect(data.proveGraffiti.toString()).to.equal('0')
+        expect(data.graffitiPreImage.toString()).to.equal('0')
+    })
+
+    it('should fail to prove zero rep', async () => {
+        const id = new ZkIdentity()
+        const epoch = 1028
+        const attesterId = 10210
+        const nonce = 0
+        const circuitInputs = genReputationCircuitInput({
+            id,
+            epoch,
+            nonce,
+            attesterId,
+            startBalance: { posRep: 10, negRep: 5 },
+            proveZeroRep: 0,
+        })
+        try {
+            await genProofAndVerify(Circuit.proveReputation, circuitInputs)
+            expect(false).to.be.true
+        } catch (err) {}
+    })
+
```

Comment:
In this test it should be `proveZeroRep: 1` and the circuit should throw error
I ran it locally and actually the prove if valid
so I found that `expect(false).to.be.true` is caught in `try-catch`
```
AssertionError: expected false to be true
    at Context.<anonymous> (/Users/zhengyawen/Documents/GitHub/Unirep/packages/circuits/test/proveReputation.test.ts:192:32) {
  showDiff: true,
  actual: false,
  expected: true,
  operator: 'strictEqual'
}
```

In summary, the proof is valid (but we hope it fail) but it doesn't throw error in the test scripts.
So we should re-write all the circuit tests, for example:
```ts
        let error
        try {
            await genProofAndVerify(Circuit.proveReputation, circuitInputs)
        } catch (err) {
            error = err
        }
        expect(error).not.equal(undefined)
```

Reply:
[da8e099](https://github.com/Unirep/Unirep/pull/216/commits/da8e099a5dcd1471a42672cf881a95b2065a6e4d)